client.go

```diff
@@ -422,6 +427,7 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 			if checkErr != nil {
 				err = checkErr
 			}
+			c.HTTPClient.CloseIdleConnections()
```

Comment:
This shouldn't be here. This branch is when good requests are returned and closing idle here will make it mostly useless. You could probably safely move this up 2 lines inside check checkErr if so it's only run if we're running with an error.

Reply:
Why do we not want to close idle connections if it's successful?

Before this patch we have keepalives totally disabled. Closing idle connections in all cases preserves that behavior except when actively retrying.

Reply:
I thought the point of this patch was to leave connections alive so that future ones are faster?

Reply:
Yes, but scoped only to the library's own retries.

Reply:
Ah ok.

Reply:
This breaks compatibility with Go < v1.12, `HTTPClient.CloseIdleConnections` was [only added in v1.12](https://golang.org/doc/go1.12#net/http) 😿